internal/pkg/agent/vault/vault_options_nondarwin.go

```diff
@@ -0,0 +1,23 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build linux || windows
```

Comment:
If it's _nondarwin_, why not use:
```suggestion
//go:build !darwin
```